src/BlockTypes/ProductGallery.php

```diff
@@ -126,12 +126,14 @@ protected function render( $attributes, $content, $block ) {
 		$dialog               = ( true === $attributes['fullScreenOnClick'] && isset( $attributes['mode'] ) && 'full' !== $attributes['mode'] ) ? $this->render_dialog() : '';
 		$post_id              = $block->context['postId'] ?? '';
 		$product              = wc_get_product( $post_id );
+		$product_id           = strval( $product->get_id() );
 
 		$html = $this->inject_dialog( $content, $dialog );
 		$p    = new \WP_HTML_Tag_Processor( $html );
 
 		if ( $p->next_tag() ) {
 			$p->set_attribute( 'data-wc-interactive', true );
+			$p->set_attribute( 'data-wc-init--watch-changes-on-add-to-cart-form', 'effects.woocommerce.watchForChangesOnAddToCartForm' );
```

Comment:
Maybe we can add a check here only to output this directive when the product type is `variable`?

Reply:
Thank you for the suggestion, I just added a condition to check for that in the code 🙌